### ISSUE TITLE
Projects don't open on fresh P10S build

### ISSUE THREAD

**Issue by lyonsil** [paranext/paranext-core#1176]
**Describe the bug**
[From Sebastian](https://discord.com/channels/1064938364597436416/1289189023478321242/1289189023478321242):
After downloading Paratext-10-Studio-0.1.0:
"Open Scripture Editor" and "Open Resource Viewer" does not do anything, "Open Text Collection" opens a window with no projects to select
copying over projects from <user>.platform.bible\projects to .paratext-10-studio\projects does not seem to change anything
attaching a debugger with C:\progs\P.B\Paratext-10-Studio-0.1.0.exe --remote-debugging-port=8315 --logLevel debug --DEV_NOISY true does not reveal related log outputs on the console
after restarting Paratext 10 a few times, the WEB project shows up in the "Open Text Collection" and "Open Resource Viewer". On the next startup of Paratext 10 it did again not show up.

**Expected behavior**
Projects open properly

Notable crash in the log:
[2024-09-27 11:53:59.131] [error] 
[dotnet data provider error]
System.Exception: Setting platformScripture.includeMyParatext9Projects was null!
   at Paranext.DataProvider.Projects.ParatextProjectDataProviderFactory.StartFactory() in D:\a\paratext-10-studio\paratext-10-studio\temp-build\paranext-core\c-sharp\Projects\ParatextProjectDataProviderFactory.cs:line 35
   at Paranext.DataProvider.Projects.ProjectDataProviderFactory.Initialize() in D:\a\paratext-10-studio\paratext-10-studio\temp-build\paranext-core\c-sharp\Projects\ProjectDataProviderFactory.cs:line 27
   at Paranext.DataProvider.Program.Main() in D:\a\paratext-10-studio\paratext-10-studio\temp-build\paranext-core\c-sharp\Program.cs:line 44
   at Paranext.DataProvider.Program.Main() in D:\a\paratext-10-studio\paratext-10-studio\temp-build\paranext-core\c-sharp\Program.cs:line 67
   at Paranext.DataProvider.Program.<Main>()
[/dotnet data provider error]
[2024-09-27 11:53:59.229] [info]  'exit' event: dotnet data provider exited with code 3762504530
[2024-09-27 11:53:59.250] [info]  'close' event: dotnet data provider exited with code 3762504530

This would cause the problem seen by Sebastian.

[main.log](https://github.com/user-attachments/files/17170389/main.log)

**Comment by Sebastian-ubs** [comment]
1st attempt
(see part of the yellow warning below that no project and no resource was found)
![Screenshot 2024-11-15 171329](https://github.com/user-attachments/assets/720eafca-1cdf-44ae-a645-891c5a4943be)

2nd attempt
Saw the same today when Ian run platform from latest main
![grafik](https://github.com/user-attachments/assets/92dbe8f4-3a51-4c50-aab8-1389fbe140e2)

**Comment by Sebastian-ubs** [comment]
I see this always when running a new build / version of Paratext for the first time or when going back to an old version. On first startup no projects would load. E.g. after running the released `Paratext 10 Studio 0.1.0.exe` downloaded and ran `Paratext 10 Studio 0.1.0-commit.3621a8d.exe`
![Image](https://github.com/user-attachments/assets/2a5cf0c2-b62d-4d75-9417-39debff2f039)
Closing and reopening leads to
![Image](https://github.com/user-attachments/assets/f4c1f59f-2b6e-4a83-aa42-669277caac53)
Then opening released `Paratext 10 Studio 0.1.0.exe` 
![Image](https://github.com/user-attachments/assets/0c34645c-2c2c-4cf1-89e5-fbe23b7edb96)
Closing and reopening
![Image](https://github.com/user-attachments/assets/ae129106-905a-4187-a5be-5b8533cecc0d)

**Comment by roopa0222** [comment]
I came across this issue when I attempted to run the test for first time on a Paratext 10 Studio 0.1.0-commit.3621a8d.exe
It works as expected when we open a second time.
I have an editable WEB project. 

```
11:03:42 [extension host] NetworkObject-NetworkObjectStatusService is being resolved now
11:03:42 [extension host warning] 11:03:42.973 > Open Scripture Editor did not find any projects with matching isEditable! Would show a prompt or something if there were a dialog available to do so [at C:\Users\rupa_\AppData\Local\Temp\2peiQ959xcSHqxg7cIJYLDmtm4F\resources\app.asar\dist\extension-host\extension-host.js:8:90092]
11:03:43 [dotnet data provider] Projects loaded from C:\Users\rupa_\.paratext-10-studio\projects\Paratext 9 Projects: TPKJ,WEB,WEB_READABLE,zzz125S_2024-06-10,zzz2_2020-10-20,zzz3_2020-10-27,zzz6,zzz6_2021-08-16
11:03:43 [dotnet data provider] Paranext data provider ready!

```

**Comment by tjcouch-sil** [comment]
Looks like the original problem is that the dotnet data provider asked for the setting at such a time that the settings service had started but the extensions had not loaded their contributions into it, hence `Request failed: "Request: object:platform.settingsServiceDataProvider-data.function (4) from 1" with error message "No setting exists for key platformScripture.includeMyParatext9Projects"`.

- [ ] Make the settings service wait while the extension service is reloading contributions.

Roopa's problem seems to be the other way around: the editor extension looked for projects faster than the dotnet data provider could serve them.

- [ ] Lengthen the max wait time for some amount of time after startup or until all services are done loading?

**Comment by roopa0222** [comment]
I got the latest code and tried it a couple of times. I could not reproduce the issue, so it looks like the fix addressed it.

**Comment by roopa0222** [comment]
@tjcouch-sil : Unfortunately I had this error in today's run.

I tested it with Platform.Bible 0.3.0-commit.d795f89.exe from https://github.com/paranext/paranext-core/actions/runs/12286455485

Based on the logs just wondering if this could be related to #941 

Please find the attached log for the main.log[AutomationFAILLog_12122024.txt](https://github.com/user-attachments/files/18113533/AutomationFAILLog_12122024.txt)

**Comment by Sebastian-ubs** [comment]
✅ I was unable to reproduce the bug and actually getting projects and resources with Paratext 10 Studio [package #30](https://github.com/paranext/paratext-10-studio/actions/runs/12232814429) of 0.1.0-commit.3621a8d (from 9th of December, #1370 was merged 5th of December).
- so to me it seems at least to have improved to not occur as frequently/reliably as before.
- caution, there are multiple package runs with for the same Pt10S build number, that would pull in different Platform (paranext-core) code, so for trying with a Paratext 10 Studio build be sure to overwrite even when the filename is unchanged. I've submitted #1393

⚠️ Icons for text collection are still missing on the first run. They appeared after the second run.



![Image](https://github.com/user-attachments/assets/32a7a272-bc67-4e11-8165-c881887a0588)
